NomaiGrandPrix/SpawnPointPool.cs

```diff
@@ -54,9 +54,12 @@ private SpawnPointPool(List<SpawnPointConfig> configs)
             this._spawnPointConfigs = configs;
         }
 
-        public static SpawnPointPool FromTsv(string pathToTsv)
+        public static SpawnPointPool FromTsv(string pathToTsv, Func<SpawnPointConfig, bool> filter)
         {
-            var configs = ParseTsv(pathToTsv).Select(line => BuildSpawnPointConfig(line)).ToList();
+            var configs = ParseTsv(pathToTsv)
+                .Select(line => BuildSpawnPointConfig(line))
+                .Where(filter)
+                .ToList();
```

Comment:
we should null guard filter so that we can give it default=null here, just because I think that requiring it doesn't make a lot of sense from the perspective of the interface